content/300-guides/050-database/830-prismas-type-system.mdx

```diff
@@ -0,0 +1,152 @@
+---
+title: How to use Prisma's type system
+metaDescription: How to use Prisma's type system
+tocDepth: 2
+---
+
+<TopBlock>
+
+This guide introduces Prisma's type system and explains how to introspect existing native types in your database, and how to use types when you apply schema changes to your database with Prisma Migrate or `db push`.
+
+</TopBlock>
+
+## How does Prisma's type system work?
+
+Prisma uses _types_ to define the kind of data that a field can hold. To make it easy to get started, Prisma provides a small number of core [scalar types](/reference/api-reference/prisma-schema-reference#model-field-scalar-types) that should cover most default use cases. For example, in the blog post model below, the `title` field uses the `String` scalar type, while the `createdAt` field uses the `DateTime` scalar type:
+
+```prisma file=schema.prisma
+model Post {
+  id        Int      @id
+  title     String
+  createdAt DateTime
+}
+```
+
+Databases also have their own type system, which defines the type of value that a column can hold. Most databases provide a large number of data types to allow fine-grained control over exactly what a column can store. For example, a database might provide inbuilt support for multiple sizes of integers, or for XML data. The names of these types vary between databases. In PostgreSQL the column type for booleans is `boolean`, whereas in MySQL the `tinyint(1)` type is typically used.
+
+### Default type mappings
+
+To allow you to get started with our core scalar types, Prisma provides _default type mappings_ that map each scalar type to a default type in the underlying database. For example:
+
+- by default Prisma's `String` type gets mapped to PostgreSQL's `text` type and MySQL's `varchar` type
+- by default Prisma's `DateTime` type gets mapped to PostgreSQL's `timestamp(3)` type and SQL Server's `datetime2` type
+
+See Prisma's [database connector pages](/concepts/database-connectors) for the default type mappings for a given database. For example, [this table](/concepts/database-connectors/postgresql#prisma-migrate) gives the default type mappings for PostgreSQL. To see the default type mappings for a given database, see the [model field scalar types section]() of the Prisma schema reference. For example, [this table](/reference/api-reference/prisma-schema-reference#model-field-scalar-types) gives the default type mappings for the `String` scalar type.
```

Comment:
This paragraph is confusing to read. I know that one is from the connector, and the other from the type direction.

Actually, I think there is a mistake:
```suggestion
See Prisma's [database connector pages](/concepts/database-connectors) for the default type mappings for a given database. For example, [this table](/concepts/database-connectors/postgresql#prisma-migrate) gives the default type mappings for PostgreSQL.   
To see the default type mappings for all databases for a specific given Prisma type, see the [model field scalar types section](/reference/api-reference/prisma-schema-reference#model-field-scalar-types) of the Prisma schema reference. For example, [this table](/reference/api-reference/prisma-schema-reference#float) gives the default type mappings for the `Float` scalar type.
```